### PR TITLE
Add comparison table to website pages

### PR DESCRIPTION
**Pull request by wildcard**
…tioning

- Add Comparison component with multi-competitor feature matrix
- Create dedicated /compare page with full comparison matrix
- Add individual VS pages for SEO (Copilot CLI, Warp, Amazon Q, OpenCode)
- Add Compare link to navigation
- Include Comparison section on homepage

Growth hacker approach inspired by IcePanel's comparison strategy:
- Scannable tables with checkmarks for quick wins
- Individual VS pages target "X vs Y" search queries
- Clear CTAs throughout for conversion
- Honest positioning that highlights Caro's unique strengths

### Changes Made

-
-
-

---

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (code restructuring without changing behavior)
- [ ] Documentation update (changes to docs, comments, or guides)
- [ ] Performance improvement (makes code faster or more efficient)
- [ ] Test coverage (adds or improves tests)
- [ ] CI/CD or tooling (changes to build, release, or development tools)

---

### Code Quality

- [ ] I have run `cargo fmt --all` (code is properly formatted)
- [ ] I have run `cargo clippy -- -D warnings` (no clippy warnings)
- [ ] I have run `cargo test` (all tests pass)
- [ ] I have run `cargo audit` (no security vulnerabilities in dependencies)

### Testing

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added contract tests for new public APIs (if applicable)
- [ ] I have added integration tests for cross-module workflows (if applicable)
- [ ] I have verified performance requirements are met (if applicable)

### Documentation

- [ ] I have added rustdoc comments for new public APIs
- [ ] I have updated relevant documentation (README, specs, guides)
- [ ] I have added examples to demonstrate new functionality (if applicable)
- [ ] I have updated the CHANGELOG.md with my changes

### TDD Workflow

- [ ] I followed the Red-Green-Refactor cycle (if implementing new features)
- [ ] I wrote failing tests before implementing the solution
- [ ] I verified tests with `cargo watch -x test` during development

---

### Deprecation Plan

---

## Related Issues and Specs

- Closes #
- Related to #
- Implements spec: `specs/<!-- feature-id -->/spec.md`
- Addresses contract: `specs/<!-- feature-id -->/contracts/<!-- contract-name -->.md`

---

### Benchmarks

```
cargo bench

Before:
- Startup time: 95ms
- Validation time: 45ms

After:
- Startup time: 85ms (-10.5%)
- Validation time: 38ms (-15.6%)
```

### Binary Size

```
Before: 47.2 MB
After:  48.1 MB (+0.9 MB)
```

### Memory Usage

---

### Before

```bash
$ cmdai "list files"
Error: command generation failed
```

### After

```bash
$ cmdai "list files"
Generated command:
  find . -type f -name "*"

Execute this command? (y/N)
```

---

### Test Output

```
$ cargo test
    Finished test [unoptimized + debuginfo] target(s) in 2.34s
     Running unittests src/lib.rs (target/debug/deps/cmdai-...)

running 42 tests
test cache::tests::test_cache_manager_retrieves_model ... ok
test safety::tests::test_dangerous_pattern_detection ... ok
...
test result: ok. 42 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

### Manual Testing

- [ ] Tested on macOS (Apple Silicon / Intel)
- [ ] Tested on Linux (Ubuntu / Fedora / Arch)
- [ ] Tested on Windows (10 / 11)
- [ ] Tested with different backends (MLX / vLLM / Ollama / Mock)
- [ ] Tested edge cases and error conditions

---

### Questions for Reviewers

---

## Reviewer Checklist

- [ ] Code follows Rust best practices and project conventions
- [ ] Tests are comprehensive and follow TDD principles
- [ ] Documentation is clear and complete
- [ ] Changes align with project specifications
- [ ] Performance impact is acceptable
- [ ] Breaking changes are justified and documented
- [ ] Security implications have been considered

---

**By submitting this PR, I confirm that:**

- [ ] My code follows the style guidelines of this project (see [AGENTS.md](https://github.com/wildcard/cmdai/blob/main/AGENTS.md))
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors
- [ ] I have read and followed the [contributing guidelines](https://github.com/wildcard/cmdai/blob/main/CONTRIBUTING.md)
- [ ] I agree to the [Code of Conduct](https://github.com/wildcard/cmdai/blob/main/CODE_OF_CONDUCT.md)

---

---
## Summary by cubic
Adds a full comparison experience to the website to help users evaluate Caro vs alternatives and improve SEO. Includes a /compare hub, competitor-specific “VS” pages, and a homepage comparison section, plus a new Compare link in the nav.

- **New Features**
  - Comparison component with multi-competitor feature matrix and legend
  - /compare page with full matrix, VS cards, and CTAs
  - New VS pages: GitHub Copilot CLI, Warp, Amazon Q CLI, OpenCode
  - Navigation: added “Compare” link
  - Homepage: added Comparison section

- **Refactors**
  - Extracted shared ComparisonPageLayout for all VS pages to remove duplication

<sup>Written for commit 51728644b515cb7c988ea586fc84302517dcd1ed. Summary will update automatically on new commits.</sup>

### CODE REVIEW

File: website/src/pages/compare/opencode.astro [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        <p><strong>OpenCode</strong> is a general-purpose AI coding agent. It can edit files, refactor code, explain codebases, and run terminal commands. It's designed to be a comprehensive coding assistant.</p>
+        <p><strong>Caro</strong> intentionally does one thing well: helping you remember and safely execute shell commands. It's a specialized sub-agent, not a full coding assistant. This focus allows for deeper expertise in shell workflows and stronger safety guarantees.</p>
+      `
+    },
+    {
+      title: 'Safety Philosophy',
+      icon: '🛡️',
+      content: `
+        <p><strong>OpenCode</strong> generates commands without systematic pre-execution validation. While you can review commands before running them, there's no dedicated layer that catches dangerous patterns.</p>
+        <p><strong>Caro</strong> validates every command against 52+ pre-compiled safety patterns. Commands like <code>rm -rf /</code>, fork bombs, and system-breaking operations are blocked before they're ever shown to you. Safety isn't optional—it's built into the core.</p>
```
[the remark this answers is not in the document]
Comment (wildcard) [reply website/src/pages/compare/opencode.astro]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wildcard/caro/pull/111#discussion_r2641939622)
